src/overlays/actors/ovl_Bg_Jya_Lift/z_bg_jya_lift.c

```diff
@@ -26,24 +34,129 @@ const ActorInit Bg_Jya_Lift_InitVars = {
     (ActorFunc)BgJyaLift_Destroy,
     (ActorFunc)BgJyaLift_Update,
     (ActorFunc)BgJyaLift_Draw,
+
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/func_80899BA0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/BgJyaLift_Init.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_F4, 1400, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_F8, 1800, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_FC, 2500, ICHAIN_STOP),
+};
+
+extern s32 D_0600D7E8;
+extern Gfx D_0600CCE0[];
+
+void BgJyaLift_InitDynapoly(BgJyaLift* this, GlobalContext* globalCtx, u32 arg2, DynaPolyMoveFlag moveFlag) {
+    s32 pad1;
+    u32 localConst = 0;
+
+    DynaPolyInfo_SetActorMove(&this->dyna, moveFlag);
+    DynaPolyInfo_Alloc(arg2, (void*)&localConst);
```

Comment:
i think this was missed in the last review, i wouldnt worry about casting this to void. the type will be fixed when bg check is decompiled, so need to put the cast

Reply:
When I tried to get rid of the cast last time it didn’t match. I’ll mess with it more later. Also should I keep the “u32 localConst” as is or make it “const u32 localConst”

Reply:
no dont worry about const. and it should definitely work without a cast to void, its done in plenty of other files. if there are any issues feel free to ping me in discord or something so we can see whats up

Reply:
I figured it out I needed to do &localConst since it takes a pointer.

Reply:
ah, yeah, the & was already there so i assumed thats what you did